protocols/ipfs.ts

```diff
@@ -153,18 +176,22 @@ export class IPFSProtocol implements Protocol<Static<typeof IPFSProtocolFields>>
           '/p2p-circuit'
         ],
         announce: [
-          `/ip4/${publicIP}/tcp/${tcpPort}`,
-          `/ip4/${publicIP}/tcp/${wsPort}/ws`
+          ...(publicIP !== '0.0.0.0' ? [`/ip4/${publicIP}/tcp/${tcpPort}`] : []),
+          ...(publicIP !== '0.0.0.0' ? [`/ip4/${publicIP}/tcp/${wsPort}/ws`] : []),
+          ...(this.options.useWebRTC === true && webrtcPort !== null && publicIP !== '0.0.0.0'
+            ? [`/ip4/${publicIP}/udp/${String(webrtcPort)}/webrtc-direct`]
+            : []),
+          '/dnsaddr/bootstrap.libp2p.io/p2p/QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN'
         ]
       },
       transports: [
         tcp(),
         webSockets(),
-        ...(this.options.useWebRTC === true ? [webRTCDirect()] : [])
+        ...(this.options.useWebRTC === true ? [webRTC(), webRTCDirect(), circuitRelayTransport()] : [])
       ],
       connectionEncrypters: [noise()],
       streamMuxers: [yamux(), mplex()],
-      peerDiscovery: [bootstrap(bootstrapConfig)],
+      peerDiscovery: [mdns(), bootstrap(bootstrapConfig)],
       services: {
```

Comment:
See the docs for the kadDHT config: https://github.com/libp2p/js-libp2p/blob/f5f20aa6cf124c824912e8968b264648ed59ab02/packages/kad-dht/src/index.ts#L532-L545

I'd remove these two configs unless you have good reason to do this:

```
          clientMode: false,
          allowQueryWithZeroPeers: true
```

Reply:
Which of the two config options specifically